driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -276,7 +277,7 @@ public TypeSystem typeSystem()
     {
         if ( connectionStage == null )
         {
-            return completedFuture( false );
+            return completedWithFalse();
```

Comment:
Since `connectionStage` is initialized with `completedWithNull`, this `connectionStage == null` check can be removed or checked against `completedWithNull`.